src/mjan.c

```diff
@@ -158,6 +159,33 @@ MOVELIST *GenerateMoves(POSITION position) {
 
         See the function CreateMovelistNode in src/core/misc.c
     */
+    char* board = (char*) SafeMalloc(16 * sizeof(char));
+    char* board = generic_hash_unhash(position, board);
+
+    int player = generic_hash_turn(position);
+    char piece = (player == 1) ? 'w' : 'b';
+    
+    int directions[4] = {-4, 4, -1, 1};
+
+    for (int i = 0; i < 16; i++) {
+        if (board[i] == piece) {
+
+            for (int d = 0; d < 4; d++) {
+                int new_pos = i + directions[d]
+
+                if (new_pos >= 0 && new_pos < 16 && board[new_pos] == '-') {
+                    if (directions[d] == -1 && (i % 4 == 0)) continue;
+                    if (directions[d] == 1 && (i % 4 == 3)) continue;
+
+                    int encoded_move = ((i & 0x0F) << 4) | (new_pos & 0x0F);
```

Comment:
what are `&` operations for? `<<` leaves 0s

Reply:
ah nvm `&` necessary for `new_pos`

Reply:
Valid